modules/modules.go

```diff
@@ -380,3 +424,22 @@ func initDependencies() error {
 
 	return nil
 }
+
+// EnterSleepMode enables or disables sleep mode for all the modules.
+func EnterSleepMode(enabled bool) {
+	// Check if differs with the old state.
+	set := sleepMode.SetToIf(!enabled, enabled)
+	if !set {
+		return
+	}
```

Comment:
This can lead to weird behavior when we start using sleep on single modules.

Shouldn't we always go through all modules and set the desired sleep mode?

So, this should probably be moved after all modules are updated, so that we can wake the task scheduler.